File: verify_cinematic.js
const { chromium } = require('playwright');

(async () => {
  const browser = await chromium.launch();
  const page = await browser.newPage();

  // Navigate to app
  try {
    await page.goto('http://localhost:8081');
  } catch (e) {
    console.log('Error navigating to app:', e);
    process.exit(1);
  }

  // Wait for load
  await page.waitForTimeout(5000);

  // 1. Verify Shop Modal Styling
  console.log('Verifying Shop Modal...');
  try {
      await page.click('text=🛒');
      await page.waitForTimeout(1000);
      await page.screenshot({ path: 'verification_shop_new.png' });
      console.log('Shop Modal screenshot saved.');
      // Close Shop (click outside or X? The X is inside overlay but outside card?)
      // My code hides global X for Shop/Backpack?
      // No, I hid the header text, but the close button is still there at top right?
      // Let's check App.js close button logic.
      // <TouchableOpacity onPress={() => setActiveModal(null)} style={...}>X</TouchableOpacity>
      // It is rendered inside modalContent.
      // modalContent is full screen transparent.
      // closeButton is absolute top:10 right:10.
      // So clicking top right should close it.
      await page.mouse.click(350, 20); // Top right roughly
      await page.waitForTimeout(1000);
  } catch (e) {
      console.log('Error verifying Shop:', e);
  }

  // 2. Verify Backpack Modal Styling
  console.log('Verifying Backpack Modal...');
  try {
      await page.click('text=🎒'); // Backpack icon
      await page.waitForTimeout(1000);
      await page.screenshot({ path: 'verification_backpack_new.png' });
      console.log('Backpack Modal screenshot saved.');
      await page.mouse.click(350, 20); // Close
      await page.waitForTimeout(1000);
  } catch (e) {
      console.log('Error verifying Backpack:', e);
  }

  // 3. Verify Feed Mode
  console.log('Verifying Feed Mode...');
  try {
      await page.click('text=🐾'); // Open Profile
      await page.waitForTimeout(1000);
      await page.click('text=Nakarm'); // Feed button
      await page.waitForTimeout(1000);
      await page.screenshot({ path: 'verification_feed_mode.png' });
      console.log('Feed Mode screenshot saved.');
      await page.click('text=X'); // Close mode
      await page.waitForTimeout(1000);
  } catch (e) {
      console.log('Error verifying Feed Mode:', e);
  }

  // 4. Verify Play Mode
  console.log('Verifying Play Mode...');
  try {
      await page.click('text=🐾'); // Open Profile
      await page.waitForTimeout(1000);
      await page.click('text=Baw się'); // Play button
      await page.waitForTimeout(1000);
      await page.screenshot({ path: 'verification_play_mode.png' });
      console.log('Play Mode screenshot saved.');
      await page.click('text=X'); // Close mode
      await page.waitForTimeout(1000);
  } catch (e) {
      console.log('Error verifying Play Mode:', e);
  }

  // 5. Verify Wash Mode
  console.log('Verifying Wash Mode...');
  try {
      await page.click('text=🐾'); // Open Profile
      await page.waitForTimeout(1000);
      await page.click('text=Umyj'); // Wash button
      await page.waitForTimeout(1000);
      await page.screenshot({ path: 'verification_wash_mode.png' });
      console.log('Wash Mode screenshot saved.');
      await page.click('text=X'); // Close mode
      await page.waitForTimeout(1000);
  } catch (e) {
      console.log('Error verifying Wash Mode:', e);
  }

  // 6. Verify Sleep Mode
  console.log('Verifying Sleep Mode...');
  try {
      await page.click('text=🐾'); // Open Profile
      await page.waitForTimeout(1000);
      await page.click('text=Sen'); // Sleep button
      await page.waitForTimeout(1000);
      await page.screenshot({ path: 'verification_sleep_mode.png' });
      console.log('Sleep Mode screenshot saved.');
      await page.click('text=☀️ Obudź'); // Wake button
      await page.waitForTimeout(1000);
  } catch (e) {
      console.log('Error verifying Sleep Mode:', e);
  }

  await browser.close();
})();

File: verify_cinematic.py
from playwright.sync_api import sync_playwright
import time

def run(playwright):
    browser = playwright.chromium.launch(headless=True)
    page = browser.new_page()
    try:
        page.goto("http://localhost:8081")
        print("Page loaded")
    except Exception as e:
        print(f"Error loading page: {e}")
        return

    # Wait for load
    page.wait_for_timeout(10000)

    # 1. Verify Shop Modal
    print("Verifying Shop Modal...")
    try:
        page.get_by_text("🛒").click()
        page.wait_for_timeout(2000)
        page.screenshot(path="verification_shop_py.png")
        print("Shop screenshot saved.")
        # Close Shop
        page.get_by_text("❌").click()
        page.wait_for_timeout(2000)
    except Exception as e:
        print(f"Error verifying Shop: {e}")

    # 2. Verify Backpack Modal
    print("Verifying Backpack Modal...")
    try:
        page.get_by_text("🎒").click()
        page.wait_for_timeout(2000)
        page.screenshot(path="verification_backpack_py.png")
        print("Backpack screenshot saved.")
        page.get_by_text("❌").click()
        page.wait_for_timeout(2000)
    except Exception as e:
        print(f"Error verifying Backpack: {e}")

    # 3. Verify Feed Mode
    print("Verifying Feed Mode...")
    try:
        page.get_by_text("🐾").click()
        page.wait_for_timeout(1000)
        page.get_by_text("Nakarm").click()
        page.wait_for_timeout(2000)
        page.screenshot(path="verification_feed_py.png")
        print("Feed screenshot saved.")
        page.get_by_text("X").click()
        page.wait_for_timeout(2000)
    except Exception as e:
        print(f"Error verifying Feed: {e}")

    # 4. Verify Play Mode
    print("Verifying Play Mode...")
    try:
        page.get_by_text("🐾").click()
        page.wait_for_timeout(1000)
        page.get_by_text("Baw się").click()
        page.wait_for_timeout(2000)
        page.screenshot(path="verification_play_py.png")
        print("Play screenshot saved.")
        page.get_by_text("X").click()
        page.wait_for_timeout(2000)
    except Exception as e:
        print(f"Error verifying Play: {e}")

    # 5. Verify Wash Mode
    print("Verifying Wash Mode...")
    try:
        page.get_by_text("🐾").click()
        page.wait_for_timeout(1000)
        page.get_by_text("Umyj").click()
        page.wait_for_timeout(2000)
        page.screenshot(path="verification_wash_py.png")
        print("Wash screenshot saved.")
        page.get_by_text("X").click()
        page.wait_for_timeout(2000)
    except Exception as e:
        print(f"Error verifying Wash: {e}")

    # 6. Verify Sleep Mode
    print("Verifying Sleep Mode...")
    try:
        page.get_by_text("🐾").click()
        page.wait_for_timeout(1000)
        page.get_by_text("Sen").click()
        page.wait_for_timeout(2000)
        page.screenshot(path="verification_sleep_py.png")
        print("Sleep screenshot saved.")
        page.get_by_text("☀️ Obudź").click()
        page.wait_for_timeout(2000)
    except Exception as e:
        print(f"Error verifying Sleep: {e}")

    browser.close()

with sync_playwright() as playwright:
    run(playwright)
